fast64_internal/oot/oot_upgrade.py

```diff
@@ -35,3 +47,167 @@ def upgradeRoomHeaders(roomObj: Object, objData: OoT_ObjectData):
             upgradeObjectList(sceneLayer.objectList, objData)
     for i in range(len(altHeaders.cutsceneHeaders)):
         upgradeObjectList(altHeaders.cutsceneHeaders[i].objectList, objData)
+
+
+#####################################
+# Cutscene
+#####################################
+def transferOldDataToNew(data, oldDataToNewData: dict):
+    # conversion to the same prop type
+    # simply transfer the old data to the new one
+    # special case for rumble subprops where it's a string to int conversion
+    for oldName, newName in oldDataToNewData.items():
+        if oldName in data:
+            if newName is not None:
+                value = data[oldName]
+
+                if newName in ["rumbleSourceStrength", "rumbleDuration", "rumbleDecreaseRate"]:
+                    value = int(getEvalParams(data[oldName]), base=16)
+
+                data[newName] = value
+
+            del data[oldName]
+
+
+def convertOldDataToEnumData(data, oldDataToEnumData: list[tuple[str, str, list[tuple[str, str, str]]]]):
```

Comment:
`list[tuple[str, str, list[tuple[str, str, str]]]]`
monkaW
what about one or two dataclasses to replace some tuples?

Reply:
I hope the changes I made is what you had in mind